core/src/main/java/de/jplag/reporting/jsonfactory/ComparisonReportWriter.java

```diff
@@ -103,9 +103,13 @@ private Match convertMatchToReportMatch(JPlagComparison comparison, de.jplag.Mat
         Token startOfSecond = tokensSecond.stream().min(lineComparator).orElseThrow();
         Token endOfSecond = tokensSecond.stream().max(lineComparator).orElseThrow();
 
+        List<Token> firstTotalTokens = tokensFirst.stream().filter(x -> x.getFile() != null && x.getFile().equals(startOfFirst.getFile())).toList();
```

Comment:
You could use Objects.equals to omit the null check

Reply:
I think that's a good idea while we wait for Java to implement a null-conditional operator 😄. I've updated my branch.